conf/cache.c

```diff
@@ -102,6 +103,53 @@ M0_INTERNAL bool m0_conf_cache_contains(struct m0_conf_cache *cache,
 	return ret;
 }
 
+M0_INTERNAL struct m0_conf_obj *
+m0_conf_cache_lookup_dynamic_fid(const struct m0_conf_cache *cache,
+				 const struct m0_fid *id)
+{
+	struct m0_conf_obj *obj;
+	struct m0_conf_obj *last_obj = NULL;
+	uint32_t            cnt1;
+	uint32_t            cnt2;
+        /*
+         * For process and service fid different approach is taken to compare
+	 * FID's. Specifically for process/service fid considering base fid only
+	 * (excluding counter bit) for comparison. Counter bits are higher 32
+	 * bits of the key.
+	 *  e.g.
+	 *  |-------- Container------------| |----------------Key-------------|
+	 *
+	 *  F F F F F F F F F F F F F F F F : F F F F F F F F  F F F F F F F F
+	 *                                    |-Counter bits-|
+	 */
+	/*
+	 * TODO: Current implementation we are assuming that dynamic FID's will
+	 * always be generated in sequential manner (w.r.t counter bits), so
+	 * that we can fetch latest last dynamic FID, but It may not be in
+	 * sequential always. This scenario needs to be handled.
+	 */
+	if ((DYNAMIC_FID_CNT(id) != 0 ) &&
+	    (M0_IN(m0_fid_tget(id), ('r','s')))) {
+		cnt1 = DYNAMIC_FID_CNT(id);
+		m0_tl_for(m0_conf_cache, &cache->ca_registry, obj) {
+			if (m0_base_fid_eq(&obj->co_id, id)) {
+				last_obj = obj;
+				cnt2 = DYNAMIC_FID_CNT(&obj->co_id);
+				if (cnt1 == cnt2)
```

Comment:
This can be further simplified as,		
				if (cnt1 != (cnt2 + 1))
					continue;
				return obj;

Reply:
I tried this change, but observing m0d crash sometimes in conf_dir_new(), Hence reverted that change.